src/persist-client/src/lib.rs

```diff
@@ -85,7 +104,8 @@ impl Id {
 /// single [Location].
 #[derive(Debug)]
 pub struct Client {
-    shards: Arc<Mutex<HashMap<Id, Arc<Mutex<State>>>>>,
+    blob: Arc<dyn BlobMulti>,
```

Comment:
I don't love the `Arc<dyn Foo>` pattern here, but I didn't want to parameterize WriteHandle + ReadHandle with the BlobMulti and Consensus types. I _think_ I have an idea for how to get this back, but I didn't want to sit too long iterating locally before starting to gather feedback

Reply:
> I think I have an idea for how to get this back, but I didn't want to sit too long iterating locally before starting to gather feedback

nope, hit a significant snag. I haven't given up on it, but I have given up on making it work in the scope of this PR